### PR TITLE
Fix regression regarding implicit working directory for Gcc linker

### PR DESCRIPTION
**Pull request by lacasseio**
### Context
This PR fix https://github.com/gradle/gradle-native/issues/139.

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and [CI build status](https://builds.gradle.org/project.html?projectId=Gradle_Check&tab=projectOverview&branch_Gradle_Check=lacasseio%2Fgn-issue-139)
- [x] Verify documentation
- [x] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/gcc/GccLinker.java [hunk 1]
```diff
@@ -41,7 +41,7 @@
     @Override
     protected Action<BuildOperationQueue<CommandLineToolInvocation>> newInvocationAction(final LinkerSpec spec, List<String> args) {
         final CommandLineToolInvocation invocation = newInvocation(
-            "linking " + spec.getOutputFile().getName(), spec.getOutputFile().getParentFile(), args, spec.getOperationLogger());
+            "linking " + spec.getOutputFile().getName(), args, spec.getOperationLogger());
```
Comment (big-guy) [inline subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/gcc/GccLinker.java]: could we explicitly use the projectDir (and keep the same signature) for now?

File: subprojects/platform-native/src/integTest/groovy/org/gradle/nativeplatform/toolchain/GccToolChainIntegrationTest.groovy [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+package org.gradle.nativeplatform.toolchain
+
+import org.gradle.nativeplatform.fixtures.AbstractInstalledToolChainIntegrationSpec
+import org.gradle.nativeplatform.fixtures.RequiresInstalledToolChain
+import org.gradle.nativeplatform.fixtures.app.CHelloWorldApp
+import spock.lang.Issue
+
+import static org.gradle.nativeplatform.fixtures.ToolChainRequirement.GCC_COMPATIBLE
+
+@RequiresInstalledToolChain(GCC_COMPATIBLE)
```
Comment (big-guy) [inline subprojects/platform-native/src/integTest/groovy/org/gradle/nativeplatform/toolchain/GccToolChainIntegrationTest.groovy]: Is this specific to Gcc or should we also have something for MSVC?

Comment (lacasseio) [reply subprojects/platform-native/src/integTest/groovy/org/gradle/nativeplatform/toolchain/GccToolChainIntegrationTest.groovy]: Good point I will fix MSVC as well.

File: subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/gcc/GccLinker.java (hunk 1 above)
Comment (lacasseio) [reply subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/gcc/GccLinker.java]: How do you think we can get the projectDir at this stage? We are pretty far down the chain. We could pass it via the `LinkerSpec`, however, it feels more like a hack than anything else. I propose to leave it like this and clean up this once we get around to improving the compiler infrastructure. What do you think?

Comment (big-guy) [reply subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/gcc/GccLinker.java]: SGTM

File: subprojects/platform-native/src/integTest/groovy/org/gradle/nativeplatform/toolchain/CommonToolChainIntegrationTest.groovy [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        given:
+        buildFile << """
+apply plugin: 'c'
+
+model {
+    components {
+        main(NativeExecutableSpec) {
+            binaries.all {
+                lib library: 'hello', linkage: 'api'
+                linker.args '-Lbuild/libs/hello/static', '-lhello'
```
Comment (big-guy) [inline subprojects/platform-native/src/integTest/groovy/org/gradle/nativeplatform/toolchain/CommonToolChainIntegrationTest.groovy]: I think this needs to be something like:

```
if (toolChain in VisualCpp) {
   linker.args 'path/to/hello static library on Windows'
} else {
   linker.args '-Lbuild/libs/hello/static', '-lhello'
}
```

Comment (lacasseio) [reply subprojects/platform-native/src/integTest/groovy/org/gradle/nativeplatform/toolchain/CommonToolChainIntegrationTest.groovy]: I used `'/LIBPATH:build/libs/hello/static', 'hello.lib'` instead.

**Pull request merged**